scripts/microshift-sos-report.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SCRIPT_NAME=$(basename "$0")
+PROFILES="network,security,microshift"
+PLUGINS="container_log,crio,rpmostree,rpm"
+TEMPDIR="/tmp"
+
+function usage() {
+    echo "Collect an sos report including all relevant information for MicroShift"
+    echo "Includes"
+    echo "  profiles: ${PROFILES}"
+    echo "  plugins: ${PLUGINS}"
+    echo ""
+    echo "Usage: ${SCRIPT_NAME} [--tmp-dir TMP-DIR]"
+    echo "   --tmp-dir TMP-DIR  Temporary directory where report will be written. Defaults to ${TEMPDIR}."
+    exit 1
+}
+
+if [ $# -ge 1 ]; then
+    case $1 in
+    --tmp-dir)
+        TEMPDIR="$2"
+        ;;
+    *)
+        usage
+        ;;
+    esac
+fi
+
+if [ "$(id -u)" -ne 0 ] ; then
+    echo "The '${SCRIPT_NAME}' script must be run with the 'root' user privileges"
+    exit 1
+fi
+
+if [ ! -d "${TEMPDIR}" ]; then
+    echo "Directory ${TEMPDIR} does not exist"
+    exit 1
+fi
+
+sos report \
+  --quiet \
+  --batch \
+  --all-logs \
+  --tmp-dir "${TEMPDIR}" \
+  --profiles "${PROFILES}" \
+  --only-plugins "${PLUGINS}"
+
+chmod +r "${TEMPDIR}"/sosreport*
```

Comment:
This is a bit dangerous because we are changing permissions of all the files that start with `sosreport`. This may not be intended by the user for all sos reports in /tmp.

Reply:
Was thinking mainly in CI, but it makes sense to require further permissions update after running the script if it runs as sudo.